packages/arb-tx-aggregator/txaggregator/server.go

```diff
@@ -107,7 +107,7 @@ func NewServer(
 func prepareTransactions(txes []DecodedBatchTx) []message.BatchTx {
 	sort.SliceStable(txes, func(i, j int) bool {
 		if !bytes.Equal(txes[i].pubkey, txes[j].pubkey) {
-			return true
+			return false
```

Comment:
I think stable sort is supposed to support this sort of thing, but I agree it's not totally clear. I think it would make sense to add some more test cases to check that this works